app-header-layout/app-header-layout.html

```diff
@@ -126,7 +126,9 @@
           value: false,
           reflectToAttribute: true,
           observer: '_hasScrollingRegionChanged'
-        }
+        },
+
+        _savedPageYOffset: Number
```

Comment:
As discussed offline:

1. `_savedPageYOffset` is only applicable for document scroll - it isn't used for element scrollers.
2. I'll move it to the prototype.